rna_design/rna_synthesizer_n.py

```diff
@@ -0,0 +1,316 @@
+            min_size_connected_component_structure_constraint=min_size_connected_component_structure_constraint,
+            min_size_connected_component_unpaired_structure_constraint=min_size_connected_component_unpaired_structure_constraint)
+
+        self.designer = AntaRNAv117Designer(Cstr=Cstr,
+                                            Cseq=Cseq,
+                                            tGC=tGC,
+                                            level=level,
+                                            tGCmax=tGCmax,
+                                            tGCvar=tGCvar,
+                                            temperature=temperature,
+                                            paramFile=paramFile,
+                                            noGUBasePair=noGUBasePair,
+                                            noLBPmanagement=noLBPmanagement,
+                                            pseudoknots=pseudoknots,
+                                            usedProgram=usedProgram,
+                                            pkprogram=pkprogram,
+                                            pkparameter=pkparameter,
+                                            HotKnots_PATH=HotKnots_PATH,
+                                            strategy=strategy,
+                                            noOfColonies=noOfColonies,
+                                            output_file=output_file,
+                                            py=py,
+                                            name=name,
+                                            verbose=verbose,
+                                            output_verbose=output_verbose,
+                                            seed=seed,
+                                            improve_procedure=improve_procedure,
+                                            Resets=Resets,
+                                            ants_per_selection=ants_per_selection,
+                                            ConvergenceCount=ConvergenceCount,
+                                            antsTerConv=antsTerConv,
+                                            alpha=alpha,
+                                            beta=beta,
+                                            ER=ER,
+                                            Cstrweight=Cstrweight,
+                                            Cgcweight=Cgcweight,
+                                            Cseqweight=Cseqweight,
+                                            omega=omega,
+                                            time=time)
+
+        self.pre_processor = PreProcessor(shape_type=shape_type,
+                                          energy_range=energy_range,
+                                          max_num=max_num,
+                                          split_components=split_components)
+
+        self.estimator = SGDClassifier()
+        self.vectorizer = Vectorizer()
```

Comment:
vectorizer is parametrized, and shoud receive the parameters from the initializer:
complexity
r
d
min_r
min_d